lifemonitor/api/models/services/github.py

```diff
@@ -0,0 +1,255 @@
+        for run in workflow.get_runs(status=status_arg):
+            yield run
+
+    def get_last_test_build(self, test_instance: models.TestInstance) -> Optional[GithubTestBuild]:
+        for run in self._iter_runs(test_instance, status=self.GithubStatus.COMPLETED):
+            return GithubTestBuild(self, test_instance, run)
+        return None
+
+    def get_last_passed_test_build(self, test_instance: models.TestInstance) -> Optional[GithubTestBuild]:
+        for run in self._iter_runs(test_instance, status=self.GithubStatus.COMPLETED):
+            if run.conclusion == self.GithubConclusion.SUCCESS:
+                return GithubTestBuild(self, test_instance, run)
+        return None
+
+    def get_last_failed_test_build(self, test_instance: models.TestInstance) -> Optional[GithubTestBuild]:
+        for run in self._iter_runs(test_instance, status=self.GithubStatus.COMPLETED):
+            if run.conclusion == self.GithubConclusion.FAILURE:
+                return GithubTestBuild(self, test_instance, run)
+        return None
+
+    def get_test_builds(self, test_instance: models.TestInstance, limit=10) -> list:
+        return list(GithubTestBuild(self, test_instance, run)
+                    for run in it.islice(self._iter_runs(test_instance), limit))
+
+    def get_test_build(self, test_instance: models.TestInstance, build_number: int) -> GithubTestBuild:
+        logger.debug("Inefficient get_test_build implementation.  Rewrite me!")
+        # TODO:  We search through the runs of the workflow because there's no
+        # obvious way to istantiate a PyGithub WorkflowRun object given a build
+        # number -- but there's has to be a way.  We can easily asseble the URL
+        # of the request to directly retrive the data we need here.
+        assert isinstance(build_number, int)
+        for run in self._iter_runs(test_instance):
+            if run.id == build_number:
+                return GithubTestBuild(self, test_instance, run)
+        raise lm_exceptions.EntityNotFoundException(models.TestBuild, entity_id=build_number)
+
+    def _parse_workflow_url(self, resource: str) -> Tuple[str, str, str]:
+        """
+        Given a URL to the testing Github Worklflow, returns a tuple
+        (server, repository, workflow_id)
+        """
+        # URL identifies a Github Workflow and should have the form:
+        #   https://api.github.com/repos/crs4/life_monitor/actions/workflows/4094661
+        expected_url_msg = " Expected '/repos/{org}/{reponame}/actions/workflows/{workflow_id}'"
+        try:
+            logger.debug("Parsing testing instance resource URL '%s'", resource)
+            result = urlparse(resource)
+            server = f"{result.scheme}://{result.netloc}"
+            # simply split the path using the / delimiter.  We'll later access the parts by index
+            parts = result.path.split('/')
+            # Validate our assumptions
+            if len(parts) != 7:
```

Comment:
Change the parsing logic to accept a relative path, as in the travis and jenkins services. Actually, we should accept both for robustness (in the end, the resource is always joined to a base URI).

Reply:
Done in ac54584.